### PR TITLE
feat(ui): add an input for template data JSON

### PR DESCRIPTION
**Pull request by umer4ik**

### CODE REVIEW

File: ui/src/routes/incarnations/EditForm.tsx [hunk 1]
```diff
@@ -9,14 +9,22 @@ import { IncarnationsForm } from './Form'
...
-  repository: x.incarnationRepository,
-  targetDirectory: x.targetDirectory,
-  templateRepository: x.templateRepository ?? '',
-  templateVersion: x.templateRepositoryVersion ?? '',
-  templateData: Object.entries(x.templateData).map(([key, value]) => ({ key, value }))
-})
+const toIncarnationInput = (x: Incarnation): IncarnationInput => {
+  let templateData = '{}'
+  try {
+    templateData = JSON.stringify(x.templateData, null, 2)
```
Comment (TheEnzoMar) [inline ui/src/routes/incarnations/EditForm.tsx:15]: Just want to check. `JSON.stringify` will only throw when a value contains a circular reference or
a `BigInt` value is encountered. Is the intended outcome here to have `templateData` be `undefined` if `JSON.stringify` is unable to serialize, but not throw?  or do you want to keep the default value or `'{}'`?

**Review by TheEnzoMar — COMMENTED** [review]
I don't have much context on FoxOps, but the general approach looks good!
I left one comment for your consideration and I'm doing a second pass to be sure I didn't miss anything.

File: ui/src/routes/incarnations/Form.tsx [hunk 2]
```diff
@@ -166,35 +182,35 @@ export const IncarnationsForm = ({
...
+                  {editTemplateDataController}
+                </Hug>
+              },
+              {
+                label: <strong>Full template data (readonly)</strong>,
+                content: <Hug my={8}>
+                  <JsonEditor
+                    defaultValue={JSON.stringify(templateDataFull, null, 2)}
+                    readOnly
+                    height="max(calc(100vh - 700px), 200px)" />
```
Comment (TheEnzoMar) [inline ui/src/routes/incarnations/Form.tsx:201]: For readability why not move this into a variable like `editTemplateDataController`

**Review by TheEnzoMar — APPROVED** [review]
The implementation looks good overall.

File: ui/src/routes/incarnations/EditForm.tsx (hunk 1 above)
Comment (umer4ik) [reply ui/src/routes/incarnations/EditForm.tsx]: I think I messed here, 'cause JSON should not have BigInt

File: ui/src/routes/incarnations/Form.tsx (hunk 2 above)
Comment (umer4ik) [reply ui/src/routes/incarnations/Form.tsx:201]: sorry, I didn't get it, the editor is already in this variable

**Pull request merged**